comp/syntheticstestscheduler/impl/scheduler.go

```diff
@@ -0,0 +1,226 @@
+func newSyntheticsTestScheduler(configs *schedulerConfigs, forwarder eventplatform.Forwarder, logger log.Component, hostNameService hostname.Component, timeFunc func() time.Time, cancel context.CancelFunc) (*syntheticsTestScheduler, error) {
+	scheduler := &syntheticsTestScheduler{
+		epForwarder:                  forwarder,
+		log:                          logger,
+		hostNameService:              hostNameService,
+		state:                        runningState{tests: map[string]*runningTestState{}},
+		cancel:                       cancel,
+		workersDone:                  make(chan struct{}),
+		flushLoopDone:                make(chan struct{}),
+		syntheticsTestProcessingChan: make(chan SyntheticsTestCtx, 100),
+		TimeNowFn:                    timeFunc,
+		workers:                      configs.workers,
+		flushInterval:                configs.flushInterval,
+		generateTestResultID:         generateRandomStringUInt63,
+		runTraceroute:                runTraceroute,
+	}
+
+	// by default, sendResult delegates to the real forwarder-backed implementation
+	scheduler.sendResult = scheduler.sendSyntheticsTestResult
+
+	scheduler.ticker = time.NewTicker(scheduler.flushInterval)
+	scheduler.tickerC = scheduler.ticker.C
+
+	if err := scheduler.loadConfigsFromCache(); err != nil {
+		return nil, err
+	}
+	return scheduler, nil
+}
+
+// runningTestState represents in-memory runtime data for a scheduled test.
+type runningTestState struct {
+	cfg     common.SyntheticsTestConfig
+	lastRun time.Time
+	nextRun time.Time
+}
+
+type runningState struct {
+	mu    sync.RWMutex
+	tests map[string]*runningTestState // PublicID -> runtime state
+}
+
+// onConfigUpdate handles remote-config updates for synthetics tests.
+func (s *syntheticsTestScheduler) onConfigUpdate(updates map[string]state.RawConfig, applyStateCallback func(string, state.ApplyStatus)) {
+	s.log.Debugf("Updates received: count=%d", len(updates))
+
+	newConfig := map[string]common.SyntheticsTestConfig{}
+	for configPath, rawConfig := range updates {
+		s.log.Debugf("received config %s: %s", configPath, string(rawConfig.Config))
+		syntheticsTestCfg := common.SyntheticsTestConfig{}
+		if err := json.Unmarshal(rawConfig.Config, &syntheticsTestCfg); err != nil {
+			s.log.Warnf("skipping invalid Synthetics Test update %s: %v", configPath, err)
+			applyStateCallback(configPath, state.ApplyStatus{
+				State: state.ApplyStateError,
+				Error: "error unmarshalling payload",
+			})
+			continue
+		}
+
+		newConfig[syntheticsTestCfg.PublicID] = syntheticsTestCfg
+		s.log.Debugf("Processed config %s: %v", configPath, syntheticsTestCfg)
+
+		applyStateCallback(configPath, state.ApplyStatus{State: state.ApplyStateAcknowledged})
+	}
+
+	if err := s.updateRunningState(newConfig); err != nil {
+		s.log.Warnf("unable to persist synthetics test config: %v", err)
+		// TODO: what path should I provide if it is global to the config
+		applyStateCallback("TODO", state.ApplyStatus{
+			State: state.ApplyStateError,
```

Comment:
Could you address this TODO? I don't know how critical this is, could be a question for remote config team

Reply:
you must put the key of the map itself, the `configPath`